src/game/gamemain.ts

```diff
@@ -0,0 +1,318 @@
+      this.walls[i].pos[1] += this.wallsSpeed * dt;
+      this.walls[i].sprite.update(dt);
+
+      // Remove if offscreen
+      if (this.walls[i].pos[1] > this.canvas.height) {
+
+        this.walls[i].pos[1] = 0 - 96;
+      }
+    }
+
+    // Update all debris
+    for (let i = 0; i < this.debris.length; i++) {
+      this.debris[i].pos[1] += this.debrisSpeed * dt;
+      this.debris[i].sprite.update(dt);
+
+      // Remove if offscreen
+      if (this.debris[i].pos[1] > this.canvas.height) {
+        this.debris.splice(i, 1);
+        i--;
+      }
+    }
+  }
+
+  collides(x: number, y: number, r: number, b: number, x2: number, y2: number, r2: number, b2: number): boolean {
+    return !(r <= x2 || x > r2
+        || b <= y2 || y > b2);
+  }
+
+  boxCollides(pos: [number, number], size: [number, number], pos2: [number, number], size2: [number, number]): boolean {
+    return this.collides(pos[0], pos[1],
+      pos[0] + size[0], pos[1] + size[1],
+      pos2[0], pos2[1],
+      pos2[0] + size2[0], pos2[1] + size2[1]);
+  }
+
+  checkPlayerBounds(): void {
+    // Check bounds
+    if (this.player.pos[0] < 0) {
+      this.player.pos[0] = 0;
+    } else if (this.player.pos[0] > this.canvas.width - this.player.sprite.size[0]) {
+      this.player.pos[0] = this.canvas.width - this.player.sprite.size[0];
+    }
+
+    if (this.player.pos[1] < 0) {
+      this.player.pos[1] = 0;
+    } else if (this.player.pos[1] > this.canvas.height - this.player.sprite.size[1]) {
+      this.player.pos[1] =this.canvas.height - this.player.sprite.size[1];
+    }
+  }
+
+  checkCollisions(): void {
+    this.checkPlayerBounds();
+
+    for (let i = 0; i < this.debris.length; i++) {
+      const pos = this.debris[i].pos;
+      const size = this.debris[i].sprite.size;
+
+      if (this.boxCollides(pos, size, this.player.pos, this.player.sprite.size)) {
+        this.col++;
+      }
+    }
+  }
+
+  renderEntities(list: TObject[]): void {
+    for (let i = 0; i < list.length; i++) {
+      this.renderEntity(list[i]);
+    }
+  }
+
+  renderEntity(entity: TObject | TPlayer): void {
+    this.ctx.save();
+    this.ctx.translate(entity.pos[0], entity.pos[1]);
+    entity.sprite.render(this.ctx, this.resources);
+    this.ctx.restore();
+  }
+
+  render(): void {
+    this.ctx.fillStyle = this.bgPattern as CanvasPattern;
+    this.ctx.fillRect(0, 0, this.canvas.width, this.canvas.height);
+
+    // Render the player if the game isn't over
+    if (!this.isGameOver) {
+      this.renderEntity(this.player);
+    }
+
+    this.renderEntities(this.walls);
+    this.renderEntities(this.debris);
+  }
+
+  reset(): void {
+    this.isGameOver = false;
+    this.gameTime = 0;
+    this.score = 0;
+
+    this.walls = [];
+    this.debris = [];
+
+    this.player.pos = [ this.canvas.width / 2 - 102 - 17, this.canvas.height - 120 ];
+
+    const numberOfWalls = Math.ceil(this.canvas.height / 96) + 1;
+    for (let i = 1; i <= numberOfWalls; i++) {
```

Comment:
🧐
```suggestion
    const numberOfWalls = Math.ceil(this.canvas.height / 96) + 1;
    
    for (let i = 1; i <= numberOfWalls; i++) {
```
Циклы и условия лучше отбивать пустой строкой